src/librustc_mir/interpret/machine.rs

```diff
@@ -232,6 +242,20 @@ pub trait Machine<'mir, 'tcx>: Sized {
         id
     }
 
+    /// In Rust, thread locals are just special statics. Therefore, the compiler
+    /// uses the same code for allocating both. However, in Miri we want to have
+    /// a property that each allocation has a unique id and, therefore, we
+    /// generate a fresh allocation id for each thread. This function takes a
+    /// potentially thread local allocation id and resolves the original static
+    /// allocation id that can be used to compute the value of the static.
+    #[inline]
+    fn resolve_thread_local_allocation_id(
```

Comment:
Leave this function without a default body in the trait. The const eval engine should `bug!` out, because it should never be able to see such an allocation (though for now just returning the input id seems fine, we can crater a change to `bug!` later, as there may be way to get `const_prop` to execute such a path).

Reply:
Updated.

Reply:
Actually, I cannot put `bug!` there because this function is called for all allocation ids. I have renamed the function and updated the documentation to explain it more clearly when this function works.